### ISSUE TITLE
This may be a basic question, but what is the recommended way to use Keras when I need to do some modifications to it?

### ISSUE THREAD

**Issue by KyotoSunshine** [fchollet/keras#374]
For example, suppose I want to use keras, but I want to try a different weight initialization method, which is not included in keras. How am I supposed to use Keras in this case (without modifying the source code of keras itself) ?

**Comment by mthrok** [comment]
You can define your custom function as Theano function and pass it to model.

See my answer in https://github.com/fchollet/keras/issues/369 for objective function case. I think it's same for initialization function.

**Comment by KyotoSunshine** [comment]
Is it really same? Here is the code that gets executed on the value of init parameter:

        self.init = initializations.get(init)

It seems that the value must be a string rather than a function, isn't it?

**Comment by KyotoSunshine** [comment]
I did it like the following, please let me know if there is a better method!

def custom_initialization(shape):
    fan_in, fan_out = keras.initializations.get_fans(shape)
    return keras.utils.theano_utils.sharedX(myfunction(shape, fan_in,_fan_out))

keras.initializations.custom_initialization = custom_initialization
...
model.add(Dense(784, 128, init='custom_initialization'))

**Comment by mthrok** [comment]
Did you see the inside of the `initializations.get` ?  https://github.com/fchollet/keras/blob/master/keras/utils/generic_utils.py#L7
It checks if the given argument is string and, if it's string look for a pre-defined function, and if not, look for global variable table and use the identifier you give.

So this works and I verified it.
```python
import theano
import theano.tensor as T
from keras.utils.theano_utils import sharedX, shared_zeros, shared_ones
def custom_initialization(shape, scale=0.05):
    '''just another normal'''
    return sharedX(np.random.randn(*shape) * scale)

model = Sequential()
model.add(Convolution2D(32, 1, 3, 3, border_mode='full', init=custom_initialization))
```

**Comment by KyotoSunshine** [comment]
Ah, I see. Looks very nice!

**Comment by mthrok** [comment]
If you ignore my advice without trying and ask for people to clean up your code, then please ask it in other place.

**Comment by KyotoSunshine** [comment]
Well, I am sorry to make you feel that way, but in reality I did not ignore your advice, I went ahead and looked into the initialization.py to see if it will work. I incorrectly assumed that the "get" method is the standard python dictionary method so it can only accept string. So, it wrongly appeared to me that your method will not work.
And the only reason I did it was that you yourself said "I think it's same for initialization function.", so it seemed that you don't know for sure. If you said it more confidently I would trust it more.
